src/attributes.cpp

```diff
@@ -2061,18 +2062,18 @@ namespace attributes {
                 ostr() << ");" << std::endl;
                 ostr() << "        }" << std::endl;
 
-                ostr() << "        if (__result.inherits(\"interrupted-error\"))"
+                ostr() << "        if (RCPP_result_gen_.inherits(\"interrupted-error\"))"
                        << std::endl
                        << "            throw Rcpp::internal::InterruptedException();"
                        << std::endl;
-                ostr() << "        if (__result.inherits(\"try-error\"))"
+                ostr() << "        if (RCPP_result_gen_.inherits(\"try-error\"))"
                        << std::endl
                        << "            throw Rcpp::exception(as<std::string>("
-                       << "__result).c_str());"
+                       << "RCPP_result_gen_).c_str());"
                        << std::endl;
                 if (!function.type().isVoid()) {
                     ostr() << "        return Rcpp::as<" << function.type() << " >"
-                           << "(__result);" << std::endl;
+                           << "(RCPP_result_gen_);" << std::endl;
```

Comment:
I'd say drop the trailing underscore as this convention is used elsewhere (e.g. in the attributes implementation) to indicate a C++ member variable.